substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -2164,14 +2228,7 @@ public String name() {
 
         @Override
         public String typeName() {
-            if (type != null) {
-                return toJavaName(type);
-            }
-            if (kind == JavaKind.Object) {
-                return "java.lang.Object";
-            } else {
-                return kind.getJavaName();
-            }
+            return valueType().toJavaName();
```

Comment:
`valueType()` has a check for `type != null` which means that it could potentially return `null` and that `null` value will not be handled here resulting in a crash.

Reply:
Yes, indeed. I have modified it to return an empty string if valueType() returns null.